pydarn/plotting/power.py

```diff
@@ -0,0 +1,150 @@
+
+    @classmethod
+    def plot_pwr0_statistic(cls, records: List[dict], beam_num: int = 0,
+                            compare: bool = True, frequency: float = 11000,
+                            statistical_calc: object = np.mean):
+
+        """
+        This function will calculate and plot a statistic of the lag-0
+        power of each record as a function of time.
+
+        This function applies the statistical function (ex. numpy.mean)
+        to the pwr0 vector (lag-0 power for each range) for each record
+        before plotting the results from all records chronologically.
+
+        Notes
+        -----
+        This code can be used to study background interference in rawacf data
+        when the radar has been operating in a receive-only mode such as 
+        "politescan" (cpid -3380), or during periods without any obvious 
+        coherent scatter returns from any range.
+
+        If you wish to compare the background interference associated with two
+        different frequencies then let compare=true. The frequencies will be
+        organized by the 'frequency' input. For example, if politescan ran
+        with 10.3 and 12.2 MHz then you could let 'frequency' equal 11000.
+        All records with tfreq below 11000 will be separated from those records
+        with tfreq above 11000. If compare = False then frequency is simply
+        the frequency that politescan ran with i.e 12.2 MHz exclusively.
+
+        Future Work
+        -----------
+        Allow for multi-beam comparison to get a directional
+        sense of the interference
+
+
+        Parameters
+        ----------
+        records: List[dict]
+        beam_num: int
+            The beam number with the desired data to plot
+        compare: bool
+            determines if a single frequency is use in plotting lag0
+            (compare=False) or two frequencies between a frequency
+            (compare=True)
+            default: False
+        frequency: int
+            frequency to specifically look for or split between depending
+            on the setting of compare.
+            default: 1100 kHz
+        statistical_calc: numpy object
+            numpy statistical calculation or generic min or max functions
+            e.g. numpy.std, numpy.median, numpy.min, numpy.max
+            default: numpy.mean
+
+        Raise
+        -----
+        NoDataFound
+        """
+
+        if compare:
+            # now compare frequencies separated by a frequency cutoff
+            # predefine the lists of separated records
+            low_freq_records = [record for record in records
+                                if record['tfreq'] < frequency]
+            high_freq_records = [record for record in records
+                                 if record['tfreq'] > frequency]
+
+            if len(low_freq_records) == 0 and high_freq_records == 0:
+                raise(exceptions.plot_exceptions.\
+                      NoDataFoundError('lag0', beam_num,
```

Comment:
I'm not sure that this exception can be used - we need to point out in the exception that if compare=True, then two separate frequencies need to be present, so might need to use a different error

Reply:
But if Compare=True then you have a range of frequencies to look at so really this should never happen and if not then there is  no data for the beam